web/src/games/cardtable/DeckContainer.tsx

```diff
@@ -0,0 +1,184 @@
+import React, { FunctionComponent } from 'react';
+import { Pattern } from 'gamesShared/definitions/cards';
+import { Card } from 'gamesShared/components/cards/Card';
+import { ICard, playerEnum, stageEnum, phaseEnum } from './game';
+import { Dialog, DialogContent, DialogActions, Slider, DialogTitle, Button } from '@material-ui/core';
+import css from './CardContainer.module.css';
+import { deckAssets, cardEnum } from './deals';
+
+interface ICardContainerProps {
+  collaborator?: {
+    handlers: {
+      handleFlip?: () => void;
+      handleDeal?: () => void;
+      handlePlay?: (idx: number) => void;
+      handleCrib?: (idx: number) => void;
+      handleCutDeal?: (idx: number) => void;
+      handleCutTurn?: (idx: number) => void;
+      handleRotateTurn: () => void;
+    };
+    gameState: {
+      playerID?: playerEnum;
+      currentDealer?: playerEnum;
+      stage?: stageEnum;
+      phase?: phaseEnum;
+      cutTie?: boolean;
+    };
+  };
+
+  cards: ICard[];
+  deck?: boolean;
+  flipped?: boolean;
+  concealed?: boolean;
+  turn?: boolean;
+  name: string;
+  className?: string;
+}
+function valuetext(value?: number) {
+  return `${value}`;
+}
+const DeckContainer: FunctionComponent<ICardContainerProps> = (props: ICardContainerProps) => {
+  let tileData = props.cards;
+  let view = !props.concealed && props.turn;
+  let stage: stageEnum = props.collaborator.gameState.stage;
+  let name = props.name;
+  let playerID: playerEnum = props.collaborator.gameState.playerID;
+  let currentDealer = props.collaborator.gameState.currentDealer;
+
+  const handleDeal = props.collaborator.handlers.handleDeal;
+  const handleCutDeal = props.collaborator.handlers.handleCutDeal;
+  const handleCutTurn = props.collaborator.handlers.handleCutTurn;
+  const handleRotate = props.collaborator.handlers.handleRotateTurn;
+  const [isOpen, setOpen] = React.useState(false);
+  const [selectedValue, setSelectedValue] = React.useState(19);
+
+  const isClickAllowed = (parentContainer: string): boolean => {
+    if (stage && parentContainer) {
```

Comment:
It is better to use guard statements to reduce nesting: if (!stage || !parentContainer) { return }... https://en.wikipedia.org/wiki/Guard_(computer_science)

Reply:
originally this was due to a higher abstraction serving multiple implementations of CardContainer.  I always favor single exit points with methods/functions to having multiple return statements.  I believe the downstream logic predicates on these states and additional conditions -- will look for a method to refactor.